sklearn/metrics/_classification.py

```diff
@@ -2089,55 +2130,128 @@ class after being classified as negative. This is the case when the
             f"problems, got targets of type: {y_type}"
         )
 
+    if raise_warning != "deprecated":
+        warnings.warn(
+            "`raise_warning` was deprecated in version 1.7 and will be removed "
+            "in 1.9, when an `UndefinedMetricWarning` will always raise in case of a "
+            "division by zero.",
+            FutureWarning,
+        )
+    else:
+        raise_warning = True
+        # TODO(1.9): This warning can be removed in 1.9. This is to make sure if users
+        # use the default args for `raise_warning` ("deprecated") and for
+        # `replace_undefined_by` (np.nan), they still get a FutureWarning telling them
+        # that the default return value will change:
+        if replace_undefined_by == "default":
+            warnings.warn(
+                "The default return value of `class_likelihood_ratios` in case of a "
+                "division by zero has been deprecated in 1.7 and will be changed to "
+                "the value set with the `replace_undefined_by` param in version 1.9. "
+                "Set `replace_undefined_by=np.nan` to silence this Warning.",
+                FutureWarning,
+            )
+
+    if replace_undefined_by == "default":
+        replace_undefined_by = np.nan
+
+    if isinstance(replace_undefined_by, dict):
+        msg = (
+            "The dictionary passed as `replace_undefined_by` needs come in the format "
+            "`{'LR+': `value_1`, 'LR-': `value_2`}` where the value for `LR+` ranges "
+            "from `1.0` to `np.inf` or is `np.nan` and the value for `LR-` ranges from "
+            f"`0.0` to `1.0` or is `np.nan`; got `{replace_undefined_by}`."
+        )
+        if ("LR+" in replace_undefined_by) and ("LR-" in replace_undefined_by):
+            desired_positive_likelihood_ratio = replace_undefined_by.get("LR+", None)
+            try:
+                check_scalar(
+                    desired_positive_likelihood_ratio,
+                    "positive_likelihood_ratio",
+                    target_type=(Real),
+                    min_val=1.0,
+                    include_boundaries="left",
+                )
+            except (TypeError, ValueError):
+                raise ValueError(msg)
+            desired_negative_likelihood_ratio = replace_undefined_by.get("LR-", None)
+            try:
+                check_scalar(
+                    desired_negative_likelihood_ratio,
+                    "negative_likelihood_ratio",
+                    target_type=(Real),
+                    min_val=0.0,
+                    max_val=1.0,
+                    include_boundaries="both",
+                )
+            except (TypeError, ValueError):
+                raise ValueError(msg)
+        else:
+            raise ValueError(msg)
```

Comment:
You could do a bit easier with one try/except:

```suggestion
        try:
            desired_lr_pos = replace_undefined_by.get("LR+", None)
            check_scalar(
                desired_lr_pos,
                "positive_likelihood_ratio",
                target_type=(Real),
                min_val=1.0,
                include_boundaries="left",
            )
            desired_lr_neg = replace_undefined_by.get("LR-", None)
            check_scalar(
                desired_lr_neg,
                "negative_likelihood_ratio",
                target_type=(Real),
                min_val=0.0,
                max_val=1.0,
                include_boundaries="both",
            )
        except Exception as e:
            raise ValueError(msg) from e
```

Reply:
Oh yes, that's a good improvement. :)
Done.